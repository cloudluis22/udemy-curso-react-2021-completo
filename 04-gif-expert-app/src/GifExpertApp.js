import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

// Componente Funcional.
const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    const deleteCategory = (category) => {
        if (window.confirm(`¿Desea eliminar las imágenes relacionadas con ${category}?`)) {
            setCategories(categories.filter(cats => cats !== category));
        }
    };

    return (
      <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories = {setCategories} />
            <hr></hr>

            <ol>
                {
                    categories.map( (category) => {
                        return <GifGrid
                            category={category}
                            key={category}
                            delete={deleteCategory}
                        />;
                    })
                }
            </ol>
        </>
    );

}

export default GifExpertApp ;