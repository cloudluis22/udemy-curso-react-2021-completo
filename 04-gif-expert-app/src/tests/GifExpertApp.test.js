import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'


describe('Pruebas en el componente principal <GifExpertApp />', () => {

    test('El componente debe renderizarse correctamente.', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar una lista de categorÃ­as.', () => {

        const categories = ['The Beatles', 'Pink Floyd'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })



})
