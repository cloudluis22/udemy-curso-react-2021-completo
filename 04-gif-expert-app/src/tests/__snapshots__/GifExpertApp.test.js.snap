// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente principal <GifExpertApp /> Debe mostrar una lista de categor√≠as. 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Beatles"
      delete={[Function]}
      key="The Beatles"
    />
    <GifGrid
      category="Pink Floyd"
      delete={[Function]}
      key="Pink Floyd"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente principal <GifExpertApp /> El componente debe renderizarse correctamente. 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
